sys/kern/syscalls.c

```diff
@@ -531,13 +531,26 @@ static int sys_clock_nanosleep(proc_t *p, clock_nanosleep_args_t *args,
   clockid_t clock_id = SCARG(args, clock_id);
   int flags = SCARG(args, flags);
   const timespec_t *u_rqtp = SCARG(args, rqtp);
-  timespec_t rqtp;
+  timespec_t *u_rmtp = SCARG(args, rmtp);
+  timespec_t rqtp, rmtp;
   int error;
 
-  if ((error = copyin_s(u_rqtp, rqtp)))
-    return error;
+  error = copyin_s(u_rqtp, rqtp);
+  if (error) {
+    *res = error;
```

Comment:
You don't have to set `*res`. You can just return error.

Reply:
You are right, thank you!